code/game/objects/items.dm

```diff
@@ -53,6 +52,10 @@
 	var/vending_cat = null// subcategory for vending machines.
 	var/list/dynamic_overlay[0] //For items which need to slightly alter their on-mob appearance while being worn.
 
+	var/is_shrapnel = 0
```

Comment:
You're adding variables to ALL items here which is not very good. You can get away with this variable here by just checking if `shrapnel_amount` is bigger than 0, at which point it's implied that the thing is_shrapnel anyways.